notebooks/tutorial.ipynb

```diff
@@ -0,0 +1,433 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "imaging.MotionCorrection.populate(**populate_settings)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "imaging.Segmentation.populate(**populate_settings)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "imaging.Fluorescence.populate(**populate_settings)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "imaging.Activity.populate(**populate_settings)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Query, fetch, and plot segmentations on average image"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import matplotlib.pyplot as plt\n",
+    "import numpy as np"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "average_image = (\n",
+    "    imaging.MotionCorrection.Summary & session_key & \"field_idx=0\"\n",
+    ").fetch1(\"average_image\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mask_xpix, mask_ypix = (\n",
+    "    imaging.Segmentation.Mask * imaging.MaskClassification.MaskType\n",
+    "    & session_key\n",
+    "    & \"mask_center_z=0\"\n",
+    "    & \"mask_npix > 130\"\n",
+    ").fetch(\"mask_xpix\", \"mask_ypix\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mask_image = np.zeros(np.shape(average_image), dtype=bool)\n",
+    "for xpix, ypix in zip(mask_xpix, mask_ypix):\n",
+    "    mask_image[ypix, xpix] = True"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.imshow(average_image)\n",
+    "plt.contour(mask_image, colors=\"white\", linewidths=0.5);"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Drop schemas\n",
+    "\n",
+    "+ Schemas are not typically dropped in a production workflow with real data in it. \n",
+    "+ At the developmental phase, it might be required for the table redesign.\n",
+    "+ When dropping all schemas is needed, the following is the dependency order."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# from workflow_calcium_imaging.pipeline import *\n",
+    "# imaging.schema.drop()\n",
+    "# scan.schema.drop()\n",
+    "# session.schema.drop()\n",
+    "# subject.schema.drop()\n",
+    "# lab.schema.drop()"
```

Comment:
Was cleaning up to prepare for a new run and tried this but got an error as well. Likely the drop order just needs to be adjusted.

```
OperationalError                          Traceback (most recent call last)
/tmp/ipykernel_3552/3274683739.py in <module>
      1 from workflow_calcium_imaging.pipeline import *
----> 2 imaging.schema.drop()
      3 scan.schema.drop()
      4 session.schema.drop()
      5 subject.schema.drop()

~/.local/lib/python3.7/site-packages/datajoint/schemas.py in drop(self, force)
    362             try:
    363                 self.connection.query(
--> 364                     "DROP DATABASE `{database}`".format(database=self.database)
    365                 )
    366                 logger.debug(

~/.local/lib/python3.7/site-packages/datajoint/connection.py in query(self, query, args, as_dict, suppress_warnings, reconnect)
    338         cursor = self._conn.cursor(cursor=cursor_class)
    339         try:
--> 340             self._execute_query(cursor, query, args, suppress_warnings)
    341         except errors.LostConnectionError:
    342             if not reconnect:

~/.local/lib/python3.7/site-packages/datajoint/connection.py in _execute_query(cursor, query, args, suppress_warnings)
    294                 cursor.execute(query, args)
...
    141     if errorclass is None:
    142         errorclass = InternalError if errno < 1000 else OperationalError
--> 143     raise errorclass(errno, errval)

OperationalError: (3730, "Cannot drop table '__segmentation' referenced by a foreign key constraint '__scan_level_report_ibfk_1' on table '__scan_level_report'.")
```

Found better success using an approach like this:

```python
def drop_databases(databases):
    import pymysql.err
    conn = dj.conn()

    with dj.config(safemode=False):
        for database in databases:
            schema = dj.Schema(f'{dj.config["custom"]["database.prefix"]}{database}')
            while schema.list_tables():
                for table in schema.list_tables():
                    try:
                        conn.query(f"DROP TABLE `{schema.database}`.`{table}`")
                    except pymysql.err.OperationalError:
                        print(f"Can't drop `{schema.database}`.`{table}`. Retrying...")
            schema.drop()

drop_databases(databases=['imaging_report', 'imaging', 'scan', 'session', 'subject', 'lab', 'reference'])
```

Reply:
Thanks!  I have incorporated your `drop_databases` function.